iml-gui/crate/src/server_date.rs

```diff
@@ -0,0 +1,15 @@
+#[derive(Default)]
+pub struct ServerDate {
+    date: Option<chrono::DateTime<chrono::offset::FixedOffset>>,
+}
+
+impl ServerDate {
+    pub(crate) fn set(self: &mut Self, date: chrono::DateTime<chrono::offset::FixedOffset>) {
+        self.date = Some(date);
+    }
+    pub(crate) fn timeago(&self, date: chrono::DateTime<chrono::offset::FixedOffset>) -> String {
+        self.date
+            .map(|sd| format!("{}", chrono_humanize::HumanTime::from(date - sd)))
```

Comment:
Just curious, does this work the same regardless of which value is larger? I.E. if `date > sd` or if `sd > date`

Reply:
I think no, it can be "ago", "in" depending on which is large.

Reply:
Since it always will be in the past (since these events aren't happening in the future), should we always subtract the larger number from the smaller number?

Reply:
I first wrote `sd - date` but that displayed as "in 2 weeks" :)  Then I checks [the sources](https://docs.rs/chrono-humanize/0.0.11/src/chrono_humanize/humantime.rs.html#147) and it looks like it assumes a negative duration is related to a moment in the past. Which makes sense IMHO. Thus `date - sd` seems correct.

Reply:
> Thus date - sd seems correct.

But what happens if `sd` is greater than `date` (which is possible). In that case it would show the event as happening in the future, correct?

Reply:
That would be interesting :) Yes, it will be in the future, unless the difference in small (10 secs) - then it will be "now". Maybe we should log such cases and show some fallback.

Reply:
Hrrm, nm I think it will still be ok. 

Since we are calculating offsets, the result (even if positive) should still yield an event time in the past.